### PR TITLE
Dala 825 version updates sprint 30

### PR DESCRIPTION
**Pull request by SiegfriedSobkowiak**
# Manual Maintenance Sprint 30

# Maintenance tasks (to be completed by the assignee)
## EDC
- [x] Complete manual maintenance in the EDC Repo
- [x] Release a new version `0.2.9`and replace the version number in the text here with the next version number to come.
- [x] Upgrade to the new version of the edc-client in the `settings.gradle.kts`
- [x] Upgrade to the new version of the edc-dummyserver in the `docker-compose.yml` file.
- [x] Upgrade to the new version of the edc-server in the `docker-compose.yml` file.
All above steps are part of another branch (DALA-370)
### Skipped updates
The following known issues need to be reviewed in case a compatible version is available. Add new known issues as they appear.
- [x] ktlint 0.45.2 (higher version is not compatible with jlleitschuh plugin)
- [x] primevue 3.15.0 (higher versions have issues with AutoComplete PrimeVue issue 2881)
- [x] primeicon 5.0.0 (higher version require higher version of primevue)
- [x] logback-classic and logback-core 1.2.11 (higher version break backend)
- [x] slf4j-api 1.7.36 (1.7.x is required in current setup otherwise logging breaks)
- [x] io.gitlab.arturbosch.detekt:detekt-cli 1.21.0 (Failed to compile)
- [x] org.jetbrains.kotlin:kotlin-compiler-embeddable 1.7.20 (CD runs were not possible)

### Gradle update
- [x] Execute `gradlew dependencyUpdates` to get a report on Dependencies with updates
- [x] Update `settings.gradle.kts` (for libraries), `build.gradle.kts` (for plugins) and `gradle.properties` (for jacoco and ktlint)
  Note: fasterXML is managed by spring, thus NO manual version update should be conducted
- [x] update the gradle wrapper: execute `gradle wrapper --gradle-version X.Y.Z`

### Dataland frontend
- [x] Update node version in `dataland-frontend/build.gradle.kts`
- [x] Update node packages: run the `updatepackages` script, e.g. by  `npm run updatepackages` to update versions in package.json
- [x]   Run the `updatepackagelock`, e.g. by  `npm run updatepackagelock` script to update `package-lock.json` and check for security issues
  (Known issues appeared in the past with updating Jest, openApiGenerator and Eslint).

### Dataland keycloak theme
- [x] Update node version in `dataland-keycloak/dataland_theme/login/build.gradle.kts`
- [x] Update node packages: run the `updatepackages` script, e.g. by  `npm run updatepackages` to update versions in package.json
- [x]   Run the `updatepackagelock`, e.g. by  `npm run updatepackagelock` script to update `package-lock.json` and check for security issues
  (Known issues appeared in the past with updating Jest, openApiGenerator and Eslint).

### Dockerfile updates
Update versions in the following dockerfiles
- [x] `./baseDockerfiles/cypressBaseImageDockerfile`
  - [x] On any change run the corresponding job in GitHub
- [x] `./baseDockerfiles/temurinBaseImageDockerfile`
  - [x] On any change run the corresponding job in GitHub
- [x] `./dataland-backend/Dockerfile`
- [x] `./dataland-keycloak/Dockerfile`
- [x] `./dataland-pgadmin/Dockerfile`
- [x] `./dataland-csvconverter/Dockerfile`
- [x] Search for all Dockerfiles that use a nginx image and update it. Find them by searching for `FROM nginx` in the whole code.
- [x] Update the versions of all postgres images in the `docker-compose.yml` file
- [x] Update the version of the node image in the `docker-compose.yml` file
- [x] Check if there are any services in the `docker-compose.yml` file that have not gotten an update yet (e.g. a new service that is not covered by the tasks above)

## Server updates
Execute `sudo apt-get update && sudo apt-get upgrade` on
- [x] dev1.dataland.com
- [x] dev2.dataland.com
- [x] tunnel.dataland.com
- [x] letsencrypt.dataland.com
- [x] (OPT) dataland.com

## Check e-mails sent by keycloak
- [x] Check that the verification e-mail is sent and displayed correctly
- [x] Check that the reset password e-mail is sent and displayed correctly
- [x] Check that the password can be reset

## Conclusion
- [x] After updating all components check if everything still works
- [x] The new version is deployed to the dev server using this branch and real data
  - [x] It's verified that this version actually is the one deployed (check actuator/info for branch name and commit id!)
  - [x] It's verified that real data has been used
  - [x] It's verified that everything seems to be working fine by manually using the website
- [x] This template has been updated to reflect the latest state of tasks required and known issues with upgrades

# Review (to be completed by the reviewer)
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github.
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described above
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed to the dev server using this branch
  - [x] It's verified that this version actually is the one deployed (check actuator/info for branch name and commit id!)
  - [x] It's verified that everything seems to be working fine by manually using the website
  - [x] All implemented Social Logins have been tested manually in the UI
  - [x] Go to the swagger-UI, authorize, run a "GET" request to the company endpoint and assure that your authorization has worked by assuring that you get a 200 response

- [x] If any work on the UI is to be merged, those changes were also documented in the Figma
- [x] The local Dev stack still works: execute `startDevelopmentStack.sh`, execute npm run cypress and run Cypress Tests locally
- [x] Locally: Go to the swagger-UI, authorize, run a "GET" request to the company endpoint and assure that your authorization has worked by assuring that you get a 200 response

### CODE REVIEW

File: dataland-backend/Dockerfile [hunk 1]
```diff
@@ -3,6 +3,6 @@ COPY . /app/
 WORKDIR /app
 RUN ./gradlew dataland-backend:bootJar --no-daemon --stacktrace
 
-FROM eclipse-temurin:17.0.4.1_1-jre-alpine
-COPY --from=builder /app/dataland-backend/build/libs/dataland-backend*.jar /jar/dataland-backend.jar
```
Comment (fsoffel) [inline dataland-backend/Dockerfile:7]: Hat das einen technischen Grund, warum du den * aufgelöst hast? Gibt es mehrere Dateien mit dem Pattern?

File: dataland-frontend/package-lock.json [hunk 2]
```diff
@@ -14,40 +14,40 @@
         "axios": "^1.1.3",
         "i18n-iso-countries": "^7.5.0",
         "json2csv": "^5.0.7",
-        "keycloak-js": "^19.0.3",
```
Comment (fsoffel) [inline dataland-frontend/package-lock.json:17]: Die Datei ist generiert, oder? 😅

File: dataland-backend/Dockerfile (hunk 1 above)
Comment (SiegfriedSobkowiak) [reply dataland-backend/Dockerfile:7]: Ja seit dem update gibt es jetzt 2 Dateien die erzeugt werden und das pattern erfüllen

File: dataland-frontend/package-lock.json (hunk 2 above)
Comment (SiegfriedSobkowiak) [reply dataland-frontend/package-lock.json:17]: ja die package lock wird generiert

Comment (fsoffel) [reply dataland-frontend/package-lock.json:17]: Phew

**Pull request merged**